node/modules/blockstore.go

```diff
@@ -45,46 +40,32 @@ func DiscardColdBlockstore(lc fx.Lifecycle, bs dtypes.UniversalBlockstore) (dtyp
 	return blockstore.NewDiscardStore(bs), nil
 }
 
-func BadgerHotBlockstore(lc fx.Lifecycle, r repo.LockedRepo) (dtypes.HotBlockstore, error) {
-	path, err := r.SplitstorePath()
+func BadgerHotBlockstore(lc fx.Lifecycle, mctx helpers.MetricsCtx, r repo.LockedRepo) (dtypes.HotBlockstore, error) {
+	bs, closer, err := r.Blockstore(helpers.LifecycleCtx(mctx, lc), repo.HotBlockstore)
 	if err != nil {
 		return nil, err
 	}
-
-	path = filepath.Join(path, "hot.badger")
-	if err := os.MkdirAll(path, 0755); err != nil {
-		return nil, err
-	}
-
-	opts, err := repo.BadgerBlockstoreOptions(repo.HotBlockstore, path, r.Readonly())
-	if err != nil {
-		return nil, err
-	}
-
-	bs, err := badgerbs.Open(opts)
-	if err != nil {
-		return nil, err
-	}
-
 	lc.Append(fx.Hook{
 		OnStop: func(_ context.Context) error {
-			return bs.Close()
-		}})
-
+			return closer()
+		},
+	})
 	return bs, nil
 }
 
-func SplitBlockstore(cfg *config.Chainstore) func(lc fx.Lifecycle, r repo.LockedRepo, ds dtypes.MetadataDS, cold dtypes.ColdBlockstore, hot dtypes.HotBlockstore) (dtypes.SplitBlockstore, error) {
-	return func(lc fx.Lifecycle, r repo.LockedRepo, ds dtypes.MetadataDS, cold dtypes.ColdBlockstore, hot dtypes.HotBlockstore) (dtypes.SplitBlockstore, error) {
+func SplitBlockstore(cfg *config.Chainstore) func(lc fx.Lifecycle, r repo.LockedRepo, ds dtypes.MetadataDS, cold dtypes.ColdBlockstore, hot dtypes.HotBlockstore) (dtypes.SplitBlockstore, func() error, error) {
```

Comment:
```suggestion
func SplitBlockstore(cfg *config.Chainstore) func(lc fx.Lifecycle, r repo.LockedRepo, ds dtypes.MetadataDS, cold dtypes.ColdBlockstore, hot dtypes.HotBlockstore) (dtypes.SplitBlockstore, error) {
```

I think this is both unnecessary and could cause a problem. It ends up passing `func(...) (dtypes.SplitBlockstore, func() error, error)` to `fx.Provide` which takes all arguments except the last (if it's an `error`) to be types to feed into the DI system to be available to match to anything that wants it. So we end up giving it both a `dtypes.SplitBlockstore` _and_ a (noop) `func() error`. I'm not sure if fx is going to use it if it's not got a named type, but it's in the DI system at least and perhaps someone ends up adding some code that requires a `func() error` and magically gets this noop one and then can't figure out why it's not doing what they expect.